src/xma/include/lib/xmahw_lib.h

```diff
@@ -95,6 +98,49 @@ typedef struct XmaHwExecBO
   }
 } XmaHwExecBO;
 
+typedef struct XmaBufferPool
+{
+    std::list<XmaBufferObj>   buffers_busy;
+    std::queue<XmaBufferObj>  buffers_free;
```

Comment:
list and queue are generally bad data structures.  Is there really a need to insert / remove from the middle here, and why are free buffers in a queue as opposed to just a vector?

Reply:
This is for tracking which buffer from pool is still available and which is already assigned for use by plugin. Can not use vector as assigned buffer can be freed out of order.

Reply:
I can understand the buffers_busy being a list for random removal, but why is buffers_free is queue?